### DISCUSSION TITLE
Testflight version 1.5.1

### DISCUSSION THREAD

**Discussion by hape42** [hape42/dailygammon#114] [General]
I have rebuilt a lot in the code. All the pages that can appear during a game, such as messages, invites, scores and much more I have outsourced in the code. On the website, this was always displayed on an extra page. I made my life harder than I had to by trying to display this so that you could still see the board. 
When you describe problems, the device and the IOS version should always be mentioned.

minimum feedback for all should be something like: "tested on iPhone XR IOS 16.1 no problems found so far".

thank you and happy testing

**Comment by kagsteiner** [comment]
Thanks, looking forward to testing it. Great reconstruction work it seems. So far only few open matches so I can't test much.

**Comment by kagsteiner** [comment]
Played a number of games on 
- iPhone 12 Pro Max
- iPad pro 12.9 inch
both with latest iOS / iPadOS
All went well, no crashes, no issues. New screeen seen that tells me current standing incl. (predicted result). Nice :-)

**Reply by Smiles2go** [reply]
iPhone 14pro, iOS 16.4.1
DG build 475

No crashes.

Only change I see is making 2 (I think) classes of messages received have gone full-screen:

1. Messages about tournaments starting or completed. These are fine, hiding a current game, since they are unrelated to current match being played. 

2. Messages about end-match. Nice that these include some details we'd not been seeing before (like "predicted" when indicated). But these do relate to match I was just playing, so having a way to see or review the associated match would be good. Also, there's again **no option for sending an end-match text message to opponent**. This was somehow lost a few updates ago, and may make DG app users appear rude by not sending customary congratulations or thank you notes. (It is still possible to send a note, but becomes a challenging multi-step process.)

**Reply by hape42** [reply]
> 2\. Messages about end-match. Nice that these include some details we'd not been seeing before (like "predicted" when indicated). But these do relate to match I was just playing, so having a way to see or review the associated match would be good. Also, there's again **no option for sending an end-match text message to opponent**. This was somehow lost a few updates ago, and may make DG app users appear rude by not sending customary congratulations or thank you notes. (It is still possible to send a note, but becomes a challenging multi-step process.)

I can only display what the server gives me in HTML.

There are two different displays from the server for end match information (sometimes also intermediate states). One is with chat and one is without chat.

![Bildschirmfoto 2023-04-29 um 08 39 50](https://user-images.githubusercontent.com/2091581/235288662-13615866-25db-4b09-ad77-5469379c3f57.png)
![IMG_0709](https://user-images.githubusercontent.com/2091581/235288699-f36eda5e-ad59-4abd-9102-8911bdd2a7ca.PNG)

I cannot transform the display without chat to the display with chat.

The buttons are mostly in a HTML form and pass a fixed URL which is composed of different information.

I can't pass a chat if that is not provided in the form.

I have to follow the flow logic of the website. An own flow logic would be extremely complex, because I would have to rebuild it to the flow logic of the server. I have tried that, but it is extremely error-prone.



Translated with www.DeepL.com/Translator (free version)

**Reply by Smiles2go** [reply]
Apologies if I misunderstood. Still, it feels like something has changed regarding end-match messages on iPhone -- seems like I get fewer offers to send messages or replies than I used to. 

No, please don't invent new send-message prompts that don't come from the html. 

Possibly consider adding a send-message button at end-match if it is easy to tell who message should go to.

**Reply by Smiles2go** [reply]
Is Gitlab working? Guess so. Couldn't save a message or upload picture yesterday. 

There are confirmed strange things with the new messaging full-screen format on iPhone. They are not severe problems, but more symptoms of DG's difficulties having no coherent and consistent approach to in-app messaging. 

It is hard to test because the new screens don't come up often, and there's no way to tell if a composed message was sent or not, received or not.

And with iOS Display Zoom is enabled, compose box falls off bottom of screen, possibly sometimes along with buttons. 

Here's screenshot of message on iPad followed by same message on iPhone 14:
![image](https://user-images.githubusercontent.com/121371957/236575070-5e30104a-34af-4c8a-8db7-87c7a3831afa.jpeg)

![image](https://user-images.githubusercontent.com/121371957/236575452-fef33282-fef8-4eab-81ae-b34b4d532954.png)

I didn't realize the black box was for message composition. 

Turning off Display Zoom works better, though when composing a message, there is no button for Send or anything else visible. Fortunately tapping outside message box caused keyboard to close and presents buttons. 

![image](https://user-images.githubusercontent.com/121371957/236575125-38b3dd92-5263-48f4-ad9b-602edcfc83be.png)

Similar issues with some messages sent by other players. I couldn't get this on two different devices or with/without Display Zoom maybe because it isn't tied to current match. So messaging was working but wasn't obvious big box within big box was for composition. No prompts or option to quote previous text as before. 

![image](https://user-images.githubusercontent.com/121371957/236575766-b4ca3a28-cb97-453a-811f-a7b1f69a2d26.png)

So the full-screen messaging and compose works on iPhone, even in Display Zoom mode. Some prompts (that don't become part of the message) would help "Enter message here..." or "Enter reply here..." depending on situation. For replies, re-add "Quote received text" button. Add a "Done Editing" button when keyboard is up. 

If text has been composed, maybe change button names to "Send then Next Game" and "Send then To Top". Is usually some way to elect to not send a composed note, so maybe add explicit "Send" and "Don't Send" buttons is better.

(What happens if outbound message was composed and user hits hamburger button in top-right instead of a button at bottom?)

**Reply by hape42** [reply]
sorry to see that.

I understand your problem with the iPhone, large font, keyboard, text box, etc.

It is on the iPhone in landscape mode now just very little space. I have also just looked at WhatsApp. They also have a similar problem as me. When the keyboard is displayed, there is very little space left for a text field in landscape mode.


Unfortunately, I must admit that I can not solve this satisfactorily.

I still program mostly alone. My capacities are not sufficient to dedicate myself to this topic adequately.

Of course I don't want to disadvantage anyone with a visual impairment.

But the topic is so big for me, I will not be able to solve it satisfactorily.

The app was originally programmed by me for the ipad. Adapting it to the iPhone causes me more problems than I ever thought.


Maybe it's better to take the iPhone variant out again altogether?

**Reply by Smiles2go** [reply]
No... please don't kill iPhone support. It is still way better than the naked web browser.

Thanks for what you've done. I'm sorry if I seemed too critical.

**Reply by kagsteiner** [reply]
Totally agree with @Smiles2go - I'm playing most of my matches on the iPad, but it is totally great to be somewhere else where you only have your phone (doctor's waiting room, for example) and have a nice UI to play DG on.

**Comment by kagsteiner** [comment]
Found a bug :-)
If you do the following: 
1. Ensure that the connection to the server is not blindingly fast :-)
2. Click “lists”
3. Click “active matches” and “active tournaments” repeatedly really fast, back and forth, faster than the app can update
4. App crashes (you should have received a crash report)

Tested on iPad Pro 12.9, iPadOS 16.4.1

**Comment by kagsteiner** [comment]
Another very minor thing that is strange is: If the server is slow or I have a very long list of matches, top page acts like this:
1. show empty list
2. Show pop up „getting topPage Data“
3. List is populated and active and usable
4. pop up still stays there for seconds

In this situation I can click on a match and play, but I‘m still wondering what the app is still retrieving from the server and why the pop-up doesn‘t go away.
![IMG_0732](https://user-images.githubusercontent.com/25194620/235837200-903a0369-9714-47cf-95e0-e759808e2c44.png)

**Comment by kagsteiner** [comment]
Playing in that tournament that feullesmortes is organizing. Many many parallel games. App behaves perfectly well. :-)

**Comment by J1Moyer** [comment]
I want to help testing, this is my email address: j1moyer@aol.com

**Comment by kagsteiner** [comment]
A friend mentioned that he would like to have a more thoughtful button placement. In some situations some buttons should have the same location on the screen, so that you can just keep clicking for default actions.

Example:
- Roll Dice -> submit greedy bearoff" - would be good if the buttons would have the same location.
- Submit -> Next (ie you submit a move and then see the opponents' move) should be at the same location.
- Roll Dice -> submit force move -> same

Counter-example:
- Next (you see opponent move as first thing in next match) -> roll dice/double : the two buttons should be on different location than "next" on purpose, because there is no clear next default button and it's bad if the app would guide me into not considering doubling.

Tricky because there are so many different situations...

**Reply by hape42** [reply]
To avoid misunderstandings, we should always try to support such topics with screenshots and possibly sketches in the screenshots. A good way is also to show how it is presented on the website and what is missing in the app to make it at least as usable as on the website. 

You can open an issue on this topic to show what needs to be done in one place.

**Comment by hape42** [comment]
done
